src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -469,35 +469,53 @@ private BoundPattern BindDeconstructionPattern(DeconstructionPatternSyntax node,
                 if (elementTypes.Length != node.SubPatterns.Count && !hasErrors)
                 {
                     var location = new SourceLocation(node.SyntaxTree, new Text.TextSpan(node.OpenParenToken.SpanStart, node.CloseParenToken.Span.End - node.OpenParenToken.SpanStart));
-                    diagnostics.Add(ErrorCode.ERR_WrongNumberOfSubpatterns, location, declType.TupleElementTypes, elementTypes.Length, node.SubPatterns.Count);
+                    diagnostics.Add(ErrorCode.ERR_WrongNumberOfSubpatterns, location, declType, elementTypes.Length, node.SubPatterns.Count);
                     hasErrors = true;
                 }
                 for (int i = 0; i < node.SubPatterns.Count; i++)
                 {
+                    var subPattern = node.SubPatterns[i];
                     bool isError = i >= elementTypes.Length;
                     TypeSymbol elementType = isError ? CreateErrorType() : elementTypes[i];
-                    // PROTOTYPE(patterns2): Check that node.SubPatterns[i].NameColon?.Name corresponds to tuple element i of declType.
-                    BoundPattern boundSubpattern = BindPattern(node.SubPatterns[i].Pattern, elementType, isError, diagnostics);
+                    if (subPattern.NameColon != null)
+                    {
+                        string name = subPattern.NameColon.Name.Identifier.ValueText;
+                        FieldSymbol foundField = CheckIsTupleElement(subPattern.NameColon.Name, (NamedTypeSymbol)declType, name, i, diagnostics);
+                        // PROTOTYPE(patterns2): Should the tuple field binding for the name be stored somewhere in the node?
+
+                    }
+                    BoundPattern boundSubpattern = BindPattern(subPattern.Pattern, elementType, isError, diagnostics);
                     patterns.Add(boundSubpattern);
                 }
             }
             else
             {
                 // It is not a tuple type. Seek an appropriate Deconstruct method.
                 var inputPlaceholder = new BoundImplicitReceiver(node, declType); // A fake receiver expression to permit us to reuse binding logic
+                // PROTOTYPE(patterns2): include element names node.SubPatterns[i].NameColon?.Name in the AnalyzedArguments
```

Comment:
In regular deconstructions (assignment/declaration/foreach), the names on the LHS do not contribute to selecting a `Deconstruct` overload. It would be rather strange that names would matter in one deconstruction context, but not others. Probably needs some LDM discussion.

Reply:
It certainly does need discussion before it is changed.